security-manager/src/main/java/org/wildfly/extension/security/manager/DeploymentPermissionsResourceDefinition.java

```diff
@@ -40,27 +67,203 @@
+            .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
+            .build();
+
+    static final SimpleAttributeDefinition MODULE = new SimpleAttributeDefinitionBuilder(PERMISSION_MODULE, ModelType.STRING)
+            .setAllowNull(true)
+            .setAllowExpression(true)
+            .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
+            .build();
+
+    private static final ObjectTypeAttributeDefinition PERMISSIONS_VALUE_TYPE =
+            ObjectTypeAttributeDefinition.Builder.of("--this name is ignored--", CLASS, NAME, ACTIONS, MODULE).build();
+
+    private static final AttributeDefinition MAXIMUM_PERMISSIONS =
+            ObjectListAttributeDefinition.Builder.of(Constants.MAXIMUM_PERMISSIONS, PERMISSIONS_VALUE_TYPE)
+            .setAllowNull(true)
+            .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
+            .build();
+
+    private static final AttributeDefinition MINIMUM_PERMISSIONS =
+            ObjectListAttributeDefinition.Builder.of(Constants.MINIMUM_PERMISSIONS, PERMISSIONS_VALUE_TYPE)
+            .setAllowNull(true)
+            .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
+            .build();
+
     static final PathElement DEPLOYMENT_PERMISSIONS_PATH = PathElement.pathElement(
             Constants.DEPLOYMENT_PERMISSIONS, Constants.DEFAULT_VALUE);
 
-    static final DeploymentPermissionsResourceDefinition INSTANCE = new DeploymentPermissionsResourceDefinition();
+    private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[] {MINIMUM_PERMISSIONS, MAXIMUM_PERMISSIONS};
 
-    private static final List<? extends PersistentResourceDefinition> CHILDREN = Collections.unmodifiableList(
-            Arrays.asList(new PermissionSetResourceDefinition(Constants.MINIMUM_SET),
-                    new PermissionSetResourceDefinition(Constants.MAXIMUM_SET)));
+    static final DeploymentPermissionsResourceDefinition INSTANCE = new DeploymentPermissionsResourceDefinition();
 
     private DeploymentPermissionsResourceDefinition() {
         super(DEPLOYMENT_PERMISSIONS_PATH, SecurityManagerExtension.getResolver(Constants.DEPLOYMENT_PERMISSIONS),
-                new AbstractAddStepHandler(), ReloadRequiredRemoveStepHandler.INSTANCE);
+                new AbstractAddStepHandler(ATTRIBUTES), ReloadRequiredRemoveStepHandler.INSTANCE);
     }
 
     @Override
     public Collection<AttributeDefinition> getAttributes() {
-        return Collections.emptySet();
+        return Collections.unmodifiableCollection(Arrays.asList(ATTRIBUTES));
     }
 
-    @Override
-    public List<? extends PersistentResourceDefinition> getChildren() {
-        return CHILDREN;
+    /**
+     * A {@link PersistentResourceXMLDescription} implementation that knows how to parse/write security permission elements.
+     *
+     * From what I (Kabir) can tell it isn't possible to parse a complex list attributes mapped by child xml elements, so I am doing it myself
```

Comment:
well you are almost right :) 
Support for that was just added https://github.com/wildfly/wildfly-core/pull/891